### PR TITLE
Update Cr3 in CurrentRootPageTable struct

### PR DESCRIPTION
**Pull request by jul-sh**

### CODE REVIEW

File: oak_restricted_kernel/src/mm/page_tables.rs [hunk 1]
```diff
@@ -279,9 +283,28 @@ impl CurrentRootPageTable {
...
+    /// Safety: The new page tables must keep the identity mapping at -2GB (kernel space) intact.
+    pub unsafe fn replace(
+        &mut self,
+        pml4_frame: PhysFrame,
+        encrypted: MemoryEncryption,
+    ) -> Option<RootPageTable> {
+        // This validates any references that expect boot page tables to be valid!
+        // Safety: Caller must ensure that the new page tables are safe.
+        unsafe {
+            Efer::update(|flags| flags.insert(EferFlags::NO_EXECUTE_ENABLE));
```
Comment (andrisaar) [inline oak_restricted_kernel/src/mm/page_tables.rs:296]: This should only be done once, you don't need to do it every time you update CR3.

Comment (jul-sh) [reply oak_restricted_kernel/src/mm/page_tables.rs]: updated, thanks

File: oak_restricted_kernel/src/lib.rs [hunk 2]
```diff
@@ -172,9 +172,17 @@ pub fn start_kernel(info: &BootParams) -> ! {
...
-        let prev_page_table = PAGE_TABLES
-            .lock()
-            .replace(mm::init_paging(program_headers).unwrap());
+        let (pml4_frame, encrypted) = mm::initial_pml4(program_headers).unwrap();
+        // Safety: the new page tables keep the identity mapping at -2GB intact, so it's safe to
+        // load the new page tables.
+        let prev_page_table = unsafe { PAGE_TABLES.lock().replace(pml4_frame, encrypted) };
+        // Prevent execution code in data only memory pages.
+        // Safety: executeable memory is assumed to be appropiately marked in the page table.
+        unsafe {
```
Comment (conradgrobler) [inline oak_restricted_kernel/src/lib.rs:181]: I think this should happen before setting the page table in CR3.

**Review by conradgrobler — APPROVED** [review]

File: oak_restricted_kernel/src/lib.rs (hunk 2 above)
Comment (jul-sh) [reply oak_restricted_kernel/src/lib.rs:178]: it does currently, cr3 is not set until L185

**Pull request merged**